rust/agama-cli/src/auth.rs

```diff
@@ -0,0 +1,246 @@
+use clap::{arg, Args, Subcommand};
+use home;
+use reqwest::header::{HeaderMap, HeaderValue, CONTENT_TYPE};
+use std::fs;
+use std::fs::File;
+use std::io;
+use std::io::{BufRead, BufReader};
+use std::os::unix::fs::PermissionsExt;
+use std::path::{Path, PathBuf};
+
+const DEFAULT_JWT_FILE: &str = ".agama/agama-jwt";
+const DEFAULT_AUTH_URL: &str = "http://localhost:3000/authenticate";
+const DEFAULT_FILE_MODE: u32 = 0o600;
+
+#[derive(Subcommand, Debug)]
+pub enum AuthCommands {
+    /// Login with defined server. Result is JWT stored locally and made available to
+    /// further use. Password can be provided by commandline option, from a file or it fallbacks
+    /// into an interactive prompt.
+    Login(LoginArgs),
+    /// Release currently stored JWT
+    Logout,
+    /// Prints currently stored JWT to stdout
+    Show,
+}
+
+/// Main entry point called from agama CLI main loop
+pub async fn run(subcommand: AuthCommands) -> anyhow::Result<()> {
+    match subcommand {
+        AuthCommands::Login(options) => login(LoginArgs::proceed(options).password()?).await,
+        AuthCommands::Logout => logout(),
+        AuthCommands::Show => show(),
+    }
+}
+
+/// Reads stored JWT token and returns it
+pub fn jwt() -> anyhow::Result<String> {
```

Comment:
No need to be public.

```suggestion
fn jwt() -> anyhow::Result<String> {
```

Reply:
We can make it public when we need it to be public 😉. If we want to make it reusable, perhaps it should be in `agama-lib`.